### ISSUE TITLE
No syntax highlighting for .slint files

### ISSUE THREAD

**Issue by Memnoc** [nvim-treesitter/nvim-treesitter#5984]
### Describe the bug

Hello Team!

Context: this [PR](https://github.com/jrmoulton/tree-sitter-slint/issues/11)
TL;DR - according to the maintainers, the [slint-ui](https://slint.dev/docs) parser tree sitter points to is not maintained any more and it seems not working at all. 

Aside from what described above, I have also tried the steps reported in the official [Slint/Neovim page.](https://github.com/slint-ui/slint/blob/master/editors/README.md#neovim)

However, I get perfect highlighting in VSCode using the [official Slint plugin ](https://github.com/slint-ui/slint/tree/master/editors/vscode)

> I have just today attempted to use this project as I am starting to work with slint.
> I can successfully install the tree sitter and `lsp` via Mason and `TSInstall slint` respectively.
> 
> However, I get completely white text, no syntax highlighting, linting or formatting provided.
> I have then tried to manually override the config offered from [LazyVim](https://www.lazyvim.org/configuration/examples) to no avail.
> 
> I have tested this on three different machines with Debian, Fedora, MacOS Sonoma respectively.
> 
> just to make sure, I have tested the same code on VSCode with the slint-extension and there I get all the features and functionalities.

Let me know if I can help this in any way - thank you!

### To Reproduce

1. `:TSInstall slint`
2. `:Mason` install `slint_lsp`
3. Test on any .slint file after having restarted Neovim
4. no LSP functions: syntax highlighting, formatting, coding functions

Output of `LspInfo` on a .`slint` file

` Configured servers list: rust_analyzer, emmet_ls, lua_ls, tsserver, clangd, jsonls, eslint, taplo, slint_lsp`

### Expected behavior

Proper LSP functions like in VSCode

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (d4c1bf7ce78051b7f4a381d1508d68928512ed5f) (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v20.10.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: cc (Debian 12.2.0-14) 12.2.0
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "6.1.0-17-amd64",
  sysname = "Linux",
  version = "#1 SMP PREEMPT_DYNAMIC Debian 6.1.69-1 (2023-12-30)"
} ~

Parser/Features         H L F I J
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - diff                ✓ . . . .
  - html                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - jsdoc               ✓ . . . .
  - json                ✓ ✓ ✓ ✓ .
  - json5               ✓ . . . ✓
  - jsonc               ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - luadoc              ✓ . . . .
  - luap                ✓ . . . .
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - regex               ✓ . . . .
  - ron                 ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - slint               ✓ . . ✓ .
  - toml                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - yaml                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1692716794

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/__w/neovim/neovim/build/nvim.AppDir/usr/share/nvim"
```

### Additional context

I use [LazyVim](https://www.lazyvim.org/) but this happens in a non-distro neovim config as well

**Comment by clason** [comment]
"Slint" is not a standard filetype; you need to manually start the correct parser (or LSP, which is completely out of scope for this repo).

**Comment by Memnoc** [comment]
Not sure I follow - .slint is a file type, and tree sitter points to an LSP that is not maintained any more. Shouldn't this be fixed?

**Comment by Memnoc** [comment]
Small follow up: `LspStart` produces [lspconfig] unhandled error: ....local/share/nvim/lazy/neoconf.nvim/lua/neoconf/util.lua:54: attempt to call upvalue 'root_dir' (a nil value)

**Comment by clason** [comment]
No it's not. I think you should invest a bit of time in understanding how Vim (and by extension Neovim) works.
